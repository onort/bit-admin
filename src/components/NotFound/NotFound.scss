@import "../../styles/main.scss";

.wrapper {
  height: 100vh;
  width: 100vw;
  background-color: $lighter-bg;
  display: flex;
}

.container {
  color: $text-color-muted;
  margin: 10% auto;
  width: px-to-rem(600px);

  span {
    display: block;
  }
}

.code {
  font-size: 7.5rem;
  font-weight: 600;
  color: lighten($text-color-muted, 35);
  line-height: 1;
  letter-spacing: px-to-rem(4px);
}

.message {
  margin-top: $spacer-md;
  font-size: $font-size-lg;
  letter-spacing: 1.025px;
}

.action {
  display: inline-block;
  margin-top: $spacer-lg;
  padding: $spacer-xxsm $spacer-xsm;
  border: 2px solid lighten($orange, 20);
  border-radius: px-to-rem(6px);
  font-size: $font-size-lg;
  font-weight: 600;
  color: lighten($orange, 15);
  transition: all 0.2s;

  &:hover {
    color: lighten($orange, 20);
  }
}
