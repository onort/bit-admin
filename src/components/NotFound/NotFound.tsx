import React from "react"
import { Link } from "react-router-dom"

import styles from "./NotFound.scss"
import { Wrapper } from "../"

const NotFound: React.SFC = props => {
  return (
    <Wrapper>
      <div className={styles.container}>
        <span className={styles.code}>404</span>
        <span className={styles.message}>
          The page you're looking for cannot be found. Make sure you have
          entered the page address correctly.
        </span>
        <Link className={styles.action} to="/">
          Homepage
        </Link>
      </div>
    </Wrapper>
  )
}

export default NotFound
