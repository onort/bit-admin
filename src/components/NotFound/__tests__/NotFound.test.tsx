import React from "react"
import { shallow } from "enzyme"
import { MemoryRouter as Router } from "react-router-dom"

import NotFound from "../"

describe("<NotFound />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Router>
        <NotFound />
      </Router>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })
})
