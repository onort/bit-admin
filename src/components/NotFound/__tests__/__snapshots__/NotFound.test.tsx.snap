// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should match snapshot 1`] = `"<div class=\\"wrapper\\"><div class=\\"container\\"><span class=\\"code\\">404</span><span class=\\"message\\">The page you&#x27;re looking for cannot be found. Make sure you have entered the page address correctly.</span><a class=\\"action\\" href=\\"/\\">Homepage</a></div></div>"`;
